src/utils/global-utils.ts

```diff
@@ -67,3 +68,18 @@ export function updateGlobalVolume(
   }
   global.save();
 }
+
+export function updateOpenInterest(
+  amount: BigInt,
+  toAddress: string,
+  fromAddress: string,
+): void {
+  let global = requireGlobal();
+
+  if (toAddress == CONDITIONAL_TOKENS_ADDRESS) {
+    global.openInterest = global.openInterest.plus(amount);
+  } else if (fromAddress == CONDITIONAL_TOKENS_ADDRESS) {
+    global.openInterest = global.openInterest.minus(amount);
+  }
+  global.save();
+}
```

Comment:
this will need to be modified to not rely on the usdc contract or conditonalTokens contract to optimize indexing